### ISSUE TITLE
Version compatible with Mac OS 10.9.5?

### ISSUE THREAD

**Issue by JGPutnam** [yishn/Sabaki#101]
Hi.  I would like to use your program but would rather not upgrade my operating system to Yosemite.  Do you have a version that works with 10.9.5?

**Comment by yishn** [comment]
It seems like the minimum OS X version supported by [Electron](https://github.com/electron/electron/) is 10.9. I don't see any reason why Sabaki shouldn't work on 10.9.5... Can you describe what's happening when you start Sabaki?

**Comment by JGPutnam** [comment]
When I try to start it, I get the following error message:  "Sabaki Cannot be opened because of a problem.  Check with the developer to make sure Sabaki works with this version of OS X.  You may need to reinstall the application.  Be sure to install any available updates for the application and OS X.  This report will be sent to Apple automatically."  What follows are the details of the problem outlined below the error message itself:

Process:         Sabaki [2247]
Path:            /Users/USER/Documents/Sabaki.app/Contents/MacOS/Sabaki
Identifier:      com.electron.sabaki
Version:         ???
Code Type:       X86-64 (Native)
Parent Process:  launchd [405]
Responsible:     Sabaki [2247]
User ID:         502

Date/Time:       2016-06-09 13:11:20.717 -0400
OS Version:      Mac OS X 10.9.5 (13F34)
Report Version:  11
Anonymous UUID:  18B3FE0F-EF4D-4A5A-F083-1AA41799A736

Sleep/Wake UUID: 2C558F23-D05D-4951-8A2E-2EBECAD46287

Crashed Thread:  0

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000002, 0x0000000000000000

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @rpath/Electron Framework.framework/Electron Framework
  Referenced from: /Users/USER/Documents/Sabaki.app/Contents/MacOS/Sabaki
  Reason: no suitable image found.  Did find:
	/Users/jputnam/Documents/Sabaki.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework: file too short

Binary Images:
    0x7fff6f997000 -     0x7fff6f9ca817  dyld (239.4) <7AD43B9B-5CEA-3C7E-9836-A06909F9CA56> /usr/lib/dyld

Model: MacBookPro5,5, BootROM MBP55.00AC.B03, 2 processors, Intel Core 2 Duo, 2.53 GHz, 4 GB, SMC 1.47f2
Graphics: NVIDIA GeForce 9400M, NVIDIA GeForce 9400M, PCI, 256 MB
Memory Module: BANK 0/DIMM0, 2 GB, DDR3, 1067 MHz, 0x80CE, 0x4D34373142353637334548312D4346382020
Memory Module: BANK 1/DIMM0, 2 GB, DDR3, 1067 MHz, 0x80CE, 0x4D34373142353637334548312D4346382020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x8D), Broadcom BCM43xx 1.0 (5.106.98.100.22)
Bluetooth: Version 4.2.7f4 15112, 3 services, 23 devices, 1 incoming serial ports
Network Service: AirPort, AirPort, en1
Serial ATA Device: Hitachi HTS545025B9SA02, 250.06 GB
Serial ATA Device: MATSHITADVD-R   UJ-868
USB Device: Iomega HDD
USB Device: Built-in iSight
USB Device: Internal Memory Card Reader
USB Device: Apple Internal Keyboard / Trackpad
USB Device: IR Receiver
USB Device: BRCM2046 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus:

**Comment by yishn** [comment]
This seems like an Electron bug. Can you try downloading the [latest version of Electron](https://github.com/electron/electron/releases) and see if it runs on your OS X? If it doesn't, it's definitely an Electron bug and I suggest you [report the issue there](https://github.com/electron/electron/). Since I don't have OS X 10.9, I'm afraid I can't do anything substantial here.

**Comment by yishn** [comment]
Oh, another thing: Electron only supports 64-bit OS X.

**Comment by JGPutnam** [comment]
Now I'm getting the following message: "Unable to find Electron app at /Applications/Sabaki.app.  Cannot find module '/Applications/Sabaki.app'"  When I try to run Sabaki from within Electron.

**Comment by yishn** [comment]
Please try to run Electron on its own without Sabaki.

**Comment by JGPutnam** [comment]
That works just fine.  Electron, the app, is fully functional.  I can get it to launch but trying to run Sabaki USING Electron does not work.

**Comment by yishn** [comment]
Yes, I understand your problem. I was just making sure that Electron itself works on your machine which it apparently does. Now we *know* that there is a problem with Sabaki. But it doesn't change the fact that I can't test Sabaki on Mac OS X 10.9.

**Comment by JGPutnam** [comment]
Do you have older versions of the app I could try out to see if they work?                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: Yichuan ShenSent: Friday, June 10, 2016 4:16 PMTo: yishn/SabakiReply To: yishn/SabakiCc: JGPutnam; AuthorSubject: Re: [yishn/Sabaki] Version compatible with Mac OS 10.9.5? (#101)Yes, I understand your problem. I was just making sure that Electron itself works on your machine which it apparently does. Now we know that there is a problem with Sabaki. But it doesn't change the fact that I can't test Sabaki on Mac OS X 10.9.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by yishn** [comment]
You can find all releases of Sabaki in the [releases page of this repository](https://github.com/yishn/Sabaki/releases).

**Comment by y-ich** [comment]
FYI
I got the similar error when I tried sabaki-v0.19.0-mac-x64.7z on macOS 10.12.1.
However, I cloned the github repository and built by npm run build:osx, then the app works fine.

**Comment by JGPutnam** [comment]
@y-ich I'm not sure I understood what you did to get the app to run correctly.  Can you walk me through the steps I should take to make Sabaki work on my mac?  Am I understanding you correctly that you got Sabaki to run after doing something to it when it wouldn't run previously?

**Comment by yishn** [comment]
I think what @y-ich did was to build Sabaki on his own, using the [build instructions](https://github.com/yishn/Sabaki/wiki/Building-&-Tests).

**Comment by JGPutnam** [comment]
@y-ich   I tried what you did and it worked!  So pumped to use it!  Thanks a bunch!

**Comment by yishn** [comment]
@y-ich and @JGPutnam, can you both try downloading the prebuilt archives of v0.19.1 and see if you still got the error message? I've built them using macOS Sierra.

**Comment by y-ich** [comment]
I confirmed that the app in sabaki-v0.19.1-mac-x64.7z starts up fine on my environment (macOS 10.12.1).
@yiishn-san, Thank you for your efforts!
And @JGPutnam-san, I am glad to hear your happy voice^^

**Comment by JGPutnam** [comment]
@yishn I tried downloading the file but it seems the "structure" of the archive is damaged.  I couldn't expand it after downloading it in zipped format

**Comment by yishn** [comment]
That's weird... @y-ich seems to have unzipped it just fine, and there are no problems on my end either... Maybe your download was corrupted; did you try it again?

**Comment by JGPutnam** [comment]
I've also confirmed that the new build (sabaki-v0.19.1-mac-x64.7z) runs on my Mac OS (10.9.5 [Mavericks])  I think I was having an issue with Stuffit expander.  In other words, Stuffit wouldn't expand the file for some reason.  I found another app that was able to do it and the program ran just fine.
